server/src-lib/Hasura/SQL/Value.hs

```diff
@@ -186,14 +187,19 @@ parsePGValue ty val = case (ty, val) of
 data TxtEncodedPGVal
   = TENull
   | TELit !Text
+  | TEJSON !Q.JSON
   deriving (Show, Eq, Generic)
 
-instance Hashable TxtEncodedPGVal
+instance Hashable TxtEncodedPGVal where
+  hashWithSalt s (TENull)            = s + (hash Null)
```

Comment:
If we can't derive this, I'd say we should delegate it to an existing hashing function, because this one looks like it would generate a lot of collisions. That is, I think it should be `hashWithSalt s = hashWithSalt s . _f` where `_f` is a function to some intermediate type we need to determine.